src/environment.cpp

```diff
@@ -762,12 +762,13 @@ class ABMHandler
 {
 private:
 	ServerEnvironment *m_env;
-	std::map<content_t, std::vector<ActiveABM> > m_aabms;
+	std::vector<std::vector<ActiveABM> *> m_aabms;
 public:
 	ABMHandler(std::vector<ABMWithState> &abms,
 			float dtime_s, ServerEnvironment *env,
 			bool use_timers):
-		m_env(env)
+		m_env(env),
+		m_aabms(65536, (std::vector<ActiveABM> *)NULL)
```

Comment:
It's the maximum possible number of node ids. As it's an u16, it won't overflow.

But I changed the code anyway, so that the magic constant is gone.